vips.go

```diff
@@ -634,6 +642,12 @@ func vipsImageType(buf []byte) ImageType {
 	if IsTypeSupported(MAGICK) && strings.HasSuffix(readImageType(buf), "MagickBuffer") {
 		return MAGICK
 	}
+	// NOTE: libheif current;y only supports heic sub types; see:
+	//   https://github.com/strukturag/libheif/issues/83#issuecomment-421427091
+	if IsTypeSupported(HEIF) && buf[4] == 0x66 && buf[5] == 0x74 && buf[6] == 0x79 && buf[7] == 0x70 &&
```

Comment:
I tried this byte sequence check in Java as below:

`return data != null && data.length > 12 && (data[4] == 0x66) && (data[5] == 0x74) && (data[6] == 0x79) && (data[7] == 0x70) &&
                (data[8] == 0x68) && (data[9] == 0x65) && (data[10] == 0x69) && (data[11] == 0x63);`

But it doesn't work for sample images at: https://github.com/nokiatech/heif_conformance/tree/master/conformance_files

Do I miss anything @jeremysf ?
Thanks.

Reply:
I think I need to update the check I wrote to support more HEIC subtypes. libheif itself doesn't support everything the HEIC container can. I think the right move here is to look at what magic / detection libheif supports and mirror that. I'll update my pull request with that when I get some time or let me know if you get it to first and what you find.

Reply:
> I think I need to update the check I wrote to support more HEIC subtypes. libheif itself doesn't support everything the HEIC container can. I think the right move here is to look at what magic / detection libheif supports and mirror that. I'll update my pull request with that when I get some time or let me know if you get it to first and what you find.

I've found a checking method which just works for me here:
http://androidxref.com/9.0.0_r3/xref/frameworks/base/media/java/android/media/ExifInterface.java#isHeifFormat